### ISSUE TITLE
Kafka 0.10.0.1 fails to start, inter.broker.protocol.version is set 0.8.2.2

### ISSUE THREAD

**Issue by atrepca** [voxpupuli/puppet-kafka#113]
### Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: `3.7.2`
- Ruby: `2.3.1p112`
- Distribution: `Ubuntu 16.04.1 LTS`
- Module version: `2.1.0`

### How to reproduce (e.g Puppet code you use)
```
class { '::kafka::broker':
  version => '0.10.0.1',
}
```

### What are you seeing
Kafka fails to start with:
```
[2016-11-01 14:39:17,668] FATAL  (kafka.Kafka$)
java.lang.IllegalArgumentException: requirement failed: log.message.format.version 0.10.0-IV1 cannot be used when inter.broker.protocol.version is set to 0.8.2.2
```

### What behaviour did you expect instead
Kafka process should start successfully.

### Output log
Pasted above.

### Any additional information you'd like to impart
This is caused by `inter.broker.protocol.version` being set to `0.8.2.2` in params.pp on [this](https://github.com/voxpupuli/puppet-kafka/blob/ec6d1a421d7ee1ff3b6864178ce1ef5583e03e92/manifests/params.pp#L124) line. Can go around it by setting the desired version in the `::kafka::broker` class, but a better alternative is to set `'inter.broker.protocol.version' => $version,` in params.pp.

**Comment by cptcanuck** [comment]
You can declare the protocol version you want when you instantiate the broker:

`
        ### default is to user inter.broker.protocol.version 8.2.2 which breaks.  Use the same version we install
        class { 'kafka::broker':
                config => { 'broker.id' => '0', 'zookeeper.connect' => 'localhost:2181', 'inter.broker.protocol.version' => '0.10.1.0' }
        }
`

**Comment by atrepca** [comment]
Yup, I mentioned that too in 'additional information'. But does it make any sense to have it hardcoded to `0.8.2.2` when it can be a param, and thus one less thing to set in `kafka::broker`? Right now you have to set the version in 2 places:
```
  class { '::kafka::broker':
    version => $version,
    config  => {
      'broker.id'                     => $broker_id,
      'zookeeper.connect'             => "my.zk.server.com:2181",
      'inter.broker.protocol.version' => $version,
    }
```
Also, `inter.broker.protocol.version` should be the current cluster version, the only time it's an older version is during rolling upgrades, as per [the Kafka docs](https://kafka.apache.org/documentation/#upgrade).

**Comment by TheMeier** [comment]
This is outdated
